consensus/src/pipeline/buffer_manager.rs

```diff
@@ -164,6 +166,10 @@ pub struct BufferManager {
     consensus_publisher: Option<Arc<ConsensusPublisher>>,
 
     pending_commit_proofs: BTreeMap<Round, LedgerInfoWithSignatures>,
+
+    // If the buffer manager receives a commit vote for a block that is not in buffer items, then
+    // the vote will be cached. We can cache upto 30 blocks, and upto 150 commit votes per block.
+    pending_commit_votes: LruCache<HashValue, Vec<CommitVote>>,
```

Comment:
this is not ideal, it's not ddos safe. a single malicious validator can flush all useful pending votes. we should maintain something similar to the PendingVote structure that only allows one vote per author.

Reply:
Done.

Reply:
I don't think this works as expected, a single malicious author can still send CommitVote for 30 fake blocks to occupy the whole cache

Reply:
Good point. Implemented a different eviction mechanism now.
If the number of blocks in pending_commit_votes exceeds 50, then out of all the blocks that were last accessed more than 100ms ago, I remove one block with least voting power. 
What do you think?

Reply:
the invariant we want to maintain is very simple - every author can only have one vote per round and we don't cache rounds that are too far. we just need to maintain a map from round -> pending votes, and each round we only allow author to have one vote. it's basically a subset of the existing PendingVotes structure.

Reply:
Yes. The current code allows only one vote per author per round.

Reply:
I don't see how the current code prevents one author to have multiple fake blocks per round that push every other blocks out

Reply:
For each block, we store `HashMap<AccountAddress, CommitVote>`. So, we can store only one commit vote per author.
In order to protect against malicious authors, we won't evict any block if the last accessed timestamp is within 100ms. When there is a need for eviction, out of all the blocks that were last accessed more than 100ms ago, we evict the blocks with least voting power. So, most likely the malicious authors' blocks will be evicted.

Reply:
this doesn't prevent a malicious author to submit millions of blocks to OOM the node since there's no limit of how many blocks it can vote for. as I said above, I think the simple way is to maintain a map of round -> pending votes that only allows one vote per author.

Reply:
Done.